aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -696,6 +692,93 @@ module AptosFramework::Stake {
         assert!(!is_current_validator(validator_1_address), 6);
     }
 
+    #[test(
+        core_framework = @0x1,
+        core_resources = @CoreResources,
+        validator_1 = @0x1,
+        validator_2 = @0x2,
+        validator_3 = @0x3,
+        validator_4 = @0x4,
+        validator_5 = @0x5
+    )]
+    public(script) fun test_validator_order(
+        core_framework: signer,
+        core_resources: signer,
+        validator_1: signer,
+        validator_2: signer,
+        validator_3: signer,
+        validator_4: signer,
+        validator_5: signer,
+    ) acquires OwnerCapability, StakePool, TestCoinCapabilities, ValidatorConfig, ValidatorSet {
+        let v1_addr = Signer::address_of(&validator_1);
+        let v2_addr = Signer::address_of(&validator_2);
+        let v3_addr = Signer::address_of(&validator_3);
+        let v4_addr = Signer::address_of(&validator_4);
+        let v5_addr = Signer::address_of(&validator_5);
+
+        Timestamp::set_time_has_started_for_testing(&core_resources);
+        initialize_validator_set(&core_resources, 100, 10000);
+
+        let (mint_cap, burn_cap) = TestCoin::initialize(&core_framework, &core_resources);
+        register_mint_stake(&validator_1, &mint_cap);
+        register_mint_stake(&validator_2, &mint_cap);
+        register_mint_stake(&validator_3, &mint_cap);
+        register_mint_stake(&validator_4, &mint_cap);
+        register_mint_stake(&validator_5, &mint_cap);
+
+        store_test_coin_mint_cap(&core_resources, mint_cap);
+        Coin::destroy_burn_cap<TestCoin>(burn_cap);
+
+        join_validator_set(&validator_3, v3_addr);
+        on_new_epoch();
+        assert!(validator_index(v3_addr) == 0, 0);
+
+        join_validator_set(&validator_4, v4_addr);
+        on_new_epoch();
+        assert!(validator_index(v3_addr) == 0, 1);
+        assert!(validator_index(v4_addr) == 1, 2);
+
+        join_validator_set(&validator_1, v1_addr);
+        on_new_epoch();
+        assert!(validator_index(v1_addr) == 0, 3);
+        assert!(validator_index(v3_addr) == 1, 4);
+        assert!(validator_index(v4_addr) == 2, 5);
+
+        join_validator_set(&validator_2, v2_addr);
+        on_new_epoch();
+        assert!(validator_index(v1_addr) == 0, 6);
+        assert!(validator_index(v2_addr) == 1, 7);
+        assert!(validator_index(v3_addr) == 2, 8);
+        assert!(validator_index(v4_addr) == 3, 9);
+
+        join_validator_set(&validator_5, v5_addr);
+        on_new_epoch();
+        assert!(validator_index(v1_addr) == 0, 10);
+        assert!(validator_index(v2_addr) == 1, 11);
+        assert!(validator_index(v3_addr) == 2, 12);
+        assert!(validator_index(v4_addr) == 3, 13);
+        assert!(validator_index(v5_addr) == 4, 14);
+    }
+
+    #[test_only]
+    public(script) fun register_mint_stake(
```

Comment:
Should we change this to `public` instead of `public(script)`? @davidiw my transpiler is grudgy about having a struct as parameter in a `public(script)`